fvm/src/engine.rs

```diff
@@ -392,17 +391,13 @@ impl Engine {
         // Note: when adding debug mode support (with recorded syscall replay) don't instrument to
         // avoid breaking debug info
 
-        use fvm_wasm_instrument::gas_metering::inject;
-        use fvm_wasm_instrument::inject_stack_limiter;
-        use fvm_wasm_instrument::parity_wasm::deserialize_buffer;
-
-        let m = deserialize_buffer(raw_wasm)?;
+        use fvm_wasm_instrument::{gas_metering, stack_limiter};
 
         // stack limiter adds post/pre-ambles to call instructions; We want to do that
         // before injecting gas accounting calls to avoid this overhead in every single
         // block of code.
-        let m =
-            inject_stack_limiter(m, self.0.config.max_wasm_stack).map_err(anyhow::Error::msg)?;
+        let raw_wasm = stack_limiter::inject(raw_wasm, self.0.config.max_wasm_stack)
```

Comment:
Do we validate the bytecode anywhere before this?

Reply:
Assuming that this will be done on actor install when wasm contracts are a thing

Reply:
We do at the top of this function as well.